packages/flame/lib/src/components/core/component.dart

```diff
@@ -259,7 +259,20 @@ class Component {
   /// This can be null if the component hasn't been added to the component tree
   /// yet, or if it is the root of component tree.
   ///
-  /// Setting this property to null is equivalent to [removeFromParent].
+  /// Setting this property to `null` is equivalent to [removeFromParent].
+  /// Setting it to a new parent component is equivalent to calling
+  /// [addToParent] and will properly remove this component from its current
+  /// parent, if any.
+  ///
+  /// Note that the [parent] setter, like [add] and similar methods,
+  /// merely enqueues the move from one parent to another. For example:
+  ///
+  /// ```dart
+  /// coin.parent = inventory;
+  /// // The inventory.children set does not include coin yet.
+  /// await game.lifecycleEventsProcessed;
```

Comment:
Added a test with a slow component.

Reply:
Does it end up in this case?
https://github.com/flame-engine/flame/blob/main/packages%2Fflame%2Flib%2Fsrc%2Fcomponents%2Fcore%2Fcomponent.dart#L811

We should be letting the lifecycle process finish and then we check again on the next tick if `onLoad` is heavy if I remember correctly, and then the future would complete before it is added, right?

Reply:
Aah right, but then it would still have lifecycle events of course and we wouldn't complete it! I wonder if there isn't a user behaviour that would lead to that the queue never empties...

Reply:
> Aah right, but then it would still have lifecycle events of course and we wouldn't complete it! I wonder if there isn't a user behaviour that would lead to that the queue never empties...

I guess it could happen if we try hard enough. Maybe if a component switches between parents on each `onMount`? But in that case, I think such a game has other problems.

We could autodetect such cases in the codebase. For example, each time the `processLifecycleEvents()` function returns with non-empty queue, increment a number, and if it reaches something high, throw an assert or a severe log. Another way would be to attach a timeout to every `lifecycleEventsProcessed` future. But I'd rather not do either since it makes things heavier and a bit more complicated than they ought to be?

Reply:
Yeah, let's keep it like it is for now, if folks starts to reporting issues with it we can figure out a solution at that point.